torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -1745,6 +1745,126 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupGloo::allgather(
 
 namespace {
 
+class AsyncAllgatherCoalescedWork : public ProcessGroupGloo::AsyncWork {
+ public:
+  AsyncAllgatherCoalescedWork(
+      const std::shared_ptr<gloo::Context>& context,
+      std::vector<std::vector<at::Tensor>>& output_lists,
+      std::vector<at::Tensor>& input_list,
+      uint32_t tag)
+      : context(context),
+        output_lists(output_lists),
+        input_list(input_list),
+        tag(tag) {}
+
+  std::shared_ptr<gloo::Context> context;
+  std::vector<std::vector<at::Tensor>> output_lists;
+  std::vector<at::Tensor> input_list;
+  const uint32_t tag;
+
+  void allgather_coalesced() {
+    assert(!output_lists.empty());
+    assert(!output_lists[0].empty());
+    assert(!input_list.empty());
```

Comment:
These checks are already done in `ProcessGroupGloo::allgather_coalesced`, any reason for doing it again here?

Reply:
This is due to linter complaining about possible out of bound access on element 0 of these vectors